src/explore-education-statistics-common/src/modules/charts/util/__tests__/createDataSetCategories.test.ts

```diff
@@ -1315,4 +1315,84 @@ describe('createDataSetCategories', () => {
       timePeriod: '2015_AY',
     });
   });
+
+  test('group by specific filter category', () => {
+    const axisConfiguration: AxisConfiguration = {
+      type: 'major',
+      groupBy: 'filters',
+      groupByFilter: 'school_type',
+      sortBy: 'name',
+      sortAsc: true,
+      dataSets: [
+        {
+          indicator: 'authorised-absence-sessions',
+          filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+        },
+        {
+          indicator: 'authorised-absence-sessions',
+          filters: ['ethnicity-major-chinese', 'state-funded-secondary'],
+        },
+      ],
+      referenceLines: [],
+      visible: true,
+      unit: '',
+      min: 0,
+    };
+
+    const fullTable = mapFullTable({
```

Comment:
So much nesting :laughing: 

Think a little improvement we could make here would be to use Immer to create a new table data with the modifications to the original.

Reply:
Nice, I didn't know about that. Much better!